kratos/sources/model_part.cpp

```diff
@@ -1758,12 +1758,21 @@ void ModelPart::RemoveSubModelPart(ModelPart& ThisSubModelPart)
 }
 
 
-ModelPart* ModelPart::GetParentModelPart() const
+ModelPart& ModelPart::GetParentModelPart()
+{
+    if (IsSubModelPart()) {
+        return *mpParentModelPart;
```

Comment:
Maybe makes sense to take the chance and modify the `ModelPart* mpParentModelPart = NULL` to `ModelPart& mrModelPart`.
By doing so the parent model part member variable will be consistent with the model one and we will avoid the dereference operators in here.

Reply:
sounds ok to me but this should be a separate PR just to be sure

Reply:
For the record. I've just realized that doing this would imply changing the way we implement the `IsSubModelPart` function (now we check the pointer is not null). Considering that checking empty references is a weird issue, I'd keep it as it is.

Reply:
I think an empty reference is undefined behavior no?
I mean dangling references is a bad thing :D

Reply:
Genau. We'd better keep the pointer :smile: